tests/e2e/tests/pilot/pilot_agent_test.go

```diff
@@ -27,3 +31,32 @@ func TestJobComplete(t *testing.T) {
 		}
 	}
 }
+
+func TestRestartEnvoy(t *testing.T) {
```

Comment:
I think this could done in an integration test without using kubernetes at all. We are trying to move away from these e2e tests

Reply:
Yeah, i am not familiar with the test framework, so it's hard for me to do that.

Reply:
Take a look at https://github.com/istio/istio/blob/master/tests/integration/README.md